### DISCUSSION TITLE
Add ESP32-C3F to GPIO Viewer

### DISCUSSION THREAD

**Discussion by lightmaster** [arendst/Tasmota#21326] [Ideas]
How can someone add new boards to the GPIO Viewer? I'd like to add ESP32-C3F. Pinout from the manufacturer can be found as the 3rd image on their Amazon listing.

Amazon listing: https://www.amazon.com/gp/product/B0BNWTCPV8/

Pinout:
![6192voW6jVL _AC_SL1001_](https://github.com/arendst/Tasmota/assets/5307827/99697240-8cad-4a49-9fa5-7a63ce3f47d2)

**Comment by sfromis** [comment]
The feature is based on the independent [GPIOViewer](https://github.com/thelastoutpostworkshop/gpio_viewer/) project, which is where you can contribute new boards.

**Reply by lightmaster** [reply]
So get it added there, and then will it be automatically pulled into Tasmota when it's updated again?

**Reply by sfromis** [reply]
Yes, when it is updated in that project.

**Reply by lightmaster** [reply]
That project's dev has added support for my board (https://github.com/thelastoutpostworkshop/gpio_viewer/issues/115). Nothing to do but to wait for Tasmota to be updated now?

**Reply by arendst** [reply]
Depends.

Chk your ``GvUrl``. If it's ``https://ota.tasmota.com/tasmota/gpioviewer/gpio_viewer_13_4_0/`` change it to ``https://thelastoutpostworkshop.github.io/microcontroller_devkit/gpio_viewer_1_5/`` and enjoy the latest changes on the lastoutpostworkshop.

To use the lastoutpostworkshop pin mode feature update to latest Tasmota dev release.

**Reply by lightmaster** [reply]
@arendst the GPIO Viewer stopped working for me today and github is saying https://thelastoutpostworkshop.github.io/microcontroller_devkit/gpio_viewer_1_5/ is "404 Not Found". How could I go about setting up my own self-hosted version of the assets needed for GvUrl?

**Reply by sfromis** [reply]
I'd suggest that you ask about what happened with the 404, and what to do, in the project page for the library used, like in the "Discussions" tab.
https://github.com/thelastoutpostworkshop/gpio_viewer/

**Reply by lightmaster** [reply]
> I'd suggest that you ask about what happened with the 404, and what to do, in the project page for the library used, like in the "Discussions" tab. https://github.com/thelastoutpostworkshop/gpio_viewer/

It's back now, but for a while there, his entire github page was coming up with a 404 error.

**Reply by sfromis** [reply]
Note that the 404 error on the specific URL you posted does not tell anything about the page working or not, as that setting is merely a prefix for accessing individual pieces of content. The server is under no "obligation" to have any content to serve if using the prefix alone, hence a 404 is entirely fair when the site is working. The version from arendst is a bit more helpful, in providing a "403 Forbidden!" instead of a 404, when lacking the resource name to be accessed under the prefix.

**Reply by lightmaster** [reply]
Fair enough, but the whole github page also had a 404 on it. Didn't know at the time that the gpio_viewer url would also have a 404 even when working.

**Comment by HofeBY** [comment]
It's probably just a temporary problem.
I just recently registered the "ESP32 C6 Zero" module.
It worked great and was extremely fast. Thanks for the great tool at

https://github.com/thelastoutpostworkshop/gpio_viewer
